pyneuroml/swc/ExportNML.py

```diff
@@ -0,0 +1,501 @@
+                ):
+                    par_seg_id = self.point_indices_vs_seg_ids[
+                        parent_point.children[0].id
+                    ]
+                    print(f"Using soma child as parent: {par_seg_id}")
+                elif (
+                    parent_point.children
+                    and parent_point.children[0].type == SWCNode.SOMA
+                    and parent_point.children[0].id == 0
+                ):
+                    par_seg_id = 0
+                    print("Using soma root as parent")
+                else:
+                    print(f"Unknown parent for segment {seg_id}: {parent_point.id}")
+                    return (
+                        f'<parent segment="UNKNOWN_PARENT_{parent_point.id}"/>'
+                        if version.startswith("2")
+                        else f' parent="UNKNOWN_PARENT_{parent_point.id}"'
+                    )
+
+                return (
+                    f'<parent segment="{par_seg_id}"/>'
+                    if version.startswith("2")
+                    else f' parent="{par_seg_id}"'
+                )
+            else:
+                par_seg_id = self.point_indices_vs_seg_ids[parent_point.id]
+                print(f"Parent segment found: {par_seg_id}")
+                return (
+                    f'<parent segment="{par_seg_id}"/>'
+                    if version.startswith("2")
+                    else f' parent="{par_seg_id}"'
+                )
+        return ""
+
+    def handle_new_cable(
+        self,
+        this_point: SWCNode,
+        cable_id: int,
+        this_type: int,
+        fract_along_parent_cable: float,
+        version: str,
+    ):
+        """
+        Handle the creation of a new cable in the NeuroML structure.
+
+        :param this_point: The current SWCNode.
+        :param cable_id: The ID of the current cable.
+        :param this_type: The type of the current point.
+        :param fract_along_parent_cable: The fractional position along the parent cable.
+        :param version: The NeuroML version being used.
+        """
+        print(
+            f"Handling new cable: Point {this_point.id}, Cable {cable_id}, Type {this_type}"
+        )
+        if version.startswith("1"):
+            fract_info = (
+                f' fract_along_parent="{fract_along_parent_cable}"'
+                if fract_along_parent_cable != 1
+                else ""
+            )
+            self.group_content.append(
+                f'      <cable id="{cable_id}" name="{self.segment_name(this_type)}_{self.seg_per_typ[this_type]}"{fract_info}>'
+            )
+            for group in self.get_groups_for_type(this_point):
+                self.group_content.append(f"        <meta:group>{group}</meta:group>")
+            self.group_content.append("      </cable>")
+        else:
+            for group in self.get_groups_for_type(this_point):
```

Comment:
if this is the first point of a new branch, it should also create a new segment group. 

ALL "cables" also have to have their own segment groups, in addition to the standard soma/dendrite etc groups. Is this being done?